src/plcc/load_spec/parse_spec/parse_semantic_spec/parse_code_fragments.py

```diff
@@ -0,0 +1,74 @@
+from dataclasses import dataclass
+from .parse_target_locator import TargetLocator, parse_target_locator
+from plcc.load_spec.load_rough_spec.parse_lines import Line
+from plcc.load_spec.load_rough_spec.parse_blocks import Block
+from plcc.load_spec.load_rough_spec.parse_dividers import Divider
+import re
+
+@dataclass
+class CodeFragment:
+    targetLocator: TargetLocator
+    block: Block
+
+def parse_code_fragments(lines_and_blocks: list[Line | Block]):
+    parser = CodeFragmentParser(lines_and_blocks)
+    return parser.parse()
+
+class CodeFragmentParser:
+    def __init__(self, lines_and_blocks: list[Line | Block]):
+        self.lines_and_blocks = lines_and_blocks
+        self.codeFragmentList = []
+        self.targetLocator = None
+        self.targetLocator_regex = r'^(\w+)(?::([a-z]+))?\s*(?:#.*)?$'
+
+    def parse(self):
+        for obj in self.lines_and_blocks:
+            self._parse_line_or_block(obj)
+
+        if self.targetLocator != None:
+            raise CodeFragmentMissingBlockError(self.targetLocator.line)
+
+        return self.codeFragmentList
+
+    def _parse_line_or_block(self, obj):
+        handler = {
+                Line: self._parse_line,
+                Block: self._parse_block
+            }.get(type(obj))
+
+        if handler:
+            handler(obj)
+
+
+    def _parse_block(self, block):
+        if self.targetLocator == None:
+            raise UndefinedTargetLocatorError(block.lines[0])
+
+        self.codeFragmentList.append(CodeFragment(targetLocator=self.targetLocator, block=block))
+        self.targetLocator = None
+
+    def _parse_line(self, line):
+        if self._isCommentOrBlank(line.string):
+                return
+
+        if self.targetLocator != None:
+            raise DuplicateTargetLocatorError(line.string)
+
+        if re.match(self.targetLocator_regex, line.string):
```

Comment:
Coverage report claims that no test covers the case when this is false. So as before, can you think of such a case. And is it based on user error or programmer error? If user, add a test. I programmer, remove the guard and add a test to demonstrate how it will fail if missused.